connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SubmittedRecords.java

```diff
@@ -261,6 +259,7 @@ static class CommittableOffsets {
         /**
          * @return the offsets that can be committed at the time of the snapshot
          */
+        @Override
         public Map<Map<String, Object>, Map<String, Object>> offsets() {
             return Collections.unmodifiableMap(offsets);
```

Comment:
we can remove `Collections.unmodifiableMap` by using `Map.copyOf` to copy `offsets` in the constructor

Reply:
I think it's actually a place where we can't use `Map.copyOf()` as some values could be null. Since the `offset` field is immutable, we can wrap it in `Collections.unmodifiableMap()` in the constructor.

Reply:
> I think it's actually a place where we can't use Map.copyOf() as some values could be null.

I hate to put `null` to the value ...

> Since the offset field is immutable, we can wrap it in Collections.unmodifiableMap() in the constructor.

+1

Reply:
Pushed an update